server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -263,16 +263,15 @@ static String generateRolloverIndexName(String sourceIndexName, IndexNameExpress
     }
 
     static void validate(MetaData metaData, RolloverRequest request) {
-        final AliasOrIndex aliasOrIndex = metaData.getAliasAndIndexLookup().get(request.getAlias());
-        if (aliasOrIndex == null) {
+        final IndexSpace indexSpace = metaData.getAliasAndIndexLookup().get(request.getAlias());
+        if (indexSpace == null) {
             throw new IllegalArgumentException("source alias does not exist");
         }
-        if (aliasOrIndex.isAlias() == false) {
+        if (indexSpace.getType() != IndexSpace.Type.ALIAS) {
             throw new IllegalArgumentException("source alias is a concrete index");
```

Comment:
Looks like a lot of error messages such as this one that assume only alias or index types would benefit from wording changes that accommodate data streams, too.

Reply:
Good point. I will go over these checks and messages and adjust.